climada/util/plot.py

```diff
@@ -874,3 +876,73 @@ def multibar_plot(ax, data, colors=None, total_width=0.8, single_width=1,
     # Draw legend if we need
     if legend:
         ax.legend(bars, data.keys())
+
+# Create GdfMeta class (as a named tuple) for GeoDataFrame meta data
+GdfMeta = namedtuple('GdfMeta', ['name', 'unit', 'col_name', 'col_unit'])
```

Comment:
hmmm I am not sure whether this is a good idea. Maybe it is, maybe not. 

@emanuel-schmid @peanutfun What do you think? Should we add some nametuple in the utility plot for a specific case of geopandas plotting?

Reply:
To add more details: this is called very generically `GdfMeta`, but refers to only one very specific data set of geodataframes.

Reply:
I suggested this to pass information to the plot that is not explicitly contained in the GeoDataFrame. `GdfMeta` can be renamed, of course. But I think a `namedtuple` makes more sense than a `dict`, because you can be sure which keys/attributes are defined. `col_name` specifies the name of the column to plot. But I agree that the names can be adjusted and made clearer.

Reply:
I like the idea of passing some meta data for the plotting. I am not sure that this is the perfect way this way :D At first I thought that `GdfMeta` some new property of `GeoDataFrames`. Also, is the current proposal general enough for a generic geopandas plotting method in CLIMADA? Or should it then be made more specific.

And, `namedtuple` makes sense indeed.

Reply:
I agree that the naming is not optimal. The included information should be: what quantity represent the values (+ units) in the gdf, what quantity represent the values (+ units) in the column labels. The rows always correspond to centroids. Maybe instead of `name` we could use `quantity` or `value_type`?

Otherwise, the plotting function should be "generic", in the sense that it produces one subplot per column of a gdf, using the values to color the different points, and using the gdf_meta varibale for figure and axis label. Of course, it is still using the geo_im_from_array util function, and I'm sure there are other ways to plot a gdf with several columns. 
Let me know what you think

Reply:
@peanutfun sorry I didn't see your suggestion in time. We can definitely add the labels and units as parameter(s) to the plot function. Then the user has to know about what their gdf represents without us helping by providing the information, which I guess is ok but is a bit more work to the user. If we put the as defaults to the plot function, the plot function is not generic anymore.

Reply:
Basically along the same lines as @peanutfun 
I'd suggest to have two parameters instead of `gdf_meta: GdfMeta`, 
`colbar_name: str` and `title_subplots: lambda`.

To get the same result as implemented now you would instead of

```python
subplots_from_gdf(
    gdf=gdf, 
    gdf_meta=GdfMeta(name='Return Period', unit='Years', col_name='Threshold Intensity', col_unit='m/s')
)
```
you would

```python
subplots_from_gdf(
    gdf=gdf, 
    colbar_name="Return Period (Years)", 
    title_subplots=lambda column: f"Threshold Intensity: {column} m/s"),
)
```

Seems better readable and has more flexibilty

Reply:
consequently, I'd let the `local_return_period` return a triple
```python
Returns
-------
gdf : GeoDataFrame
label : str
    GeoDataFrame label
column_label : lambda
    Column label
```